declarations_site/chatbot/views.py

```diff
@@ -1,31 +1,213 @@
 import re
 import json
 from random import choice
-from django.urls import reverse
 from django.conf import settings
-from django.http import HttpResponse, HttpResponseBadRequest, HttpResponseForbidden, HttpResponseNotAllowed
+from django.urls import reverse
+from django.utils.http import urlencode
 from django.views.decorators.csrf import csrf_exempt
-from spotter.utils import reverse_qs
-from chatbot.utils import ukr_plural, chat_response, simple_search, verify_jwt
+from django.http import (HttpResponse, HttpResponseBadRequest, HttpResponseForbidden,
+    HttpResponseNotAllowed)
+from chatbot.utils import (ukr_plural, chat_response, simple_search, verify_jwt, chat_last_message,
+    create_subscription, find_subscription2, list_subscriptions, load_notify)
+from spotter.utils import load_declarations
+
+
+def reverse_qs(viewname, qs=None, **kwargs):
+    url = reverse(viewname, **kwargs)
+    if qs:
+        url += '?' + urlencode(qs)
+    return url
+
+
+def botcmd_subscribe(data):
+    text = data['text']
+
+    if ' ' in text:
+        _, text = text.split(' ', 1)
+    else:
+        text = chat_last_message(data, True)
+
+    if not text or is_bot_command(text):
+        return chat_response(data, 'Спочатку зробіть запит')
+
+    try:
+        create_subscription(data, text.strip())
+    except Exception as e:
```

Comment:
I don't know all possible exceptions in DjangoStorage when creating user, so prefer catch all (in this particular case) instead of showing 500 for user.